src/plugins/visualizations/public/wizard/type_selection/type_selection.tsx

```diff
@@ -187,27 +187,31 @@ class TypeSelection extends React.Component<TypeSelectionProps, TypeSelectionSta
 
     const allTypes = [...types, ...visTypes.getAliases()];
 
-    let entries: Array<VisTypeListEntry | VisTypeAliasListEntry>;
+    let entries: VisTypeListEntry[];
     if (!query) {
-      entries = allTypes.map((type) => ({ ...type, highlighted: false }));
+      entries = allTypes.map((type) => ({ type, highlighted: false }));
     } else {
       const q = query.toLowerCase();
       entries = allTypes.map((type) => {
         const matchesQuery =
           type.name.toLowerCase().includes(q) ||
           type.title.toLowerCase().includes(q) ||
           (typeof type.description === 'string' && type.description.toLowerCase().includes(q));
-        return { ...type, highlighted: matchesQuery };
```

Comment:
:information_source: The main problem here was that we were spreading the type here and thus losing the actual `BaseVisTypeClass` that has getters on it. I changed this structure slightly, so that we now just assign the `type` as a reference into each `VisTypeListEntry`. All the further changes in this file are related to this minor change.